ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m

```diff
@@ -11,17 +11,131 @@
+                                                             RCErrorContainer * _Nullable error) {
+    }];
+    BOOL isAnonymous __unused = RCCommonFunctionality.isAnonymous;
+
+
+    [RCCommonFunctionality purchaseProduct:@""
+                   signedDiscountTimestamp:@""
+                           completionBlock:^(NSDictionary * _Nullable customerInfo,
+                                             RCErrorContainer * _Nullable error) {
+    }];
+
+     [RCCommonFunctionality purchasePackage:@""
+                                   offering:@""
+                    signedDiscountTimestamp:@""
+                            completionBlock:^(NSDictionary * _Nullable customerInfo,
+                                              RCErrorContainer * _Nullable error) {
+     }];
+
+    [RCCommonFunctionality makeDeferredPurchase:^(RCPurchaseCompletedBlock _Nonnull purchaseCompleted) {
+    }
+                                completionBlock:^(NSDictionary * _Nullable customerInfo,
+                                                  RCErrorContainer * _Nullable error) {
+    }];
+
+    [RCCommonFunctionality setFinishTransactions:NO];
+    [RCCommonFunctionality checkTrialOrIntroductoryPriceEligibility:@[@""]
+                                                    completionBlock:^(NSDictionary<NSString *,RCIntroEligibility *>
+                                                                      * _Nonnull eligibilities) {
+    }];
+     [RCCommonFunctionality paymentDiscountForProductIdentifier:@""
+                                                       discount:@""
+                                                completionBlock:^(NSDictionary * _Nullable discount,
+                                                                  RCErrorContainer * _Nullable error) {
+     }];
+
+    if (@available(iOS 14.0, *)) {
+        [RCCommonFunctionality presentCodeRedemptionSheet];
+    }
+    [RCCommonFunctionality invalidatePurchaserInfoCache];
+    [RCCommonFunctionality canMakePaymentsWithFeatures:@[]];
+}
+
+- (void)testSubscriberAttributes {
+    [RCCommonFunctionality setAttributes:@{}];
+    [RCCommonFunctionality setEmail:@""];
+    [RCCommonFunctionality setPhoneNumber:@""];
+    [RCCommonFunctionality setDisplayName:@""];
+    [RCCommonFunctionality setPushToken:@""];
+    [RCCommonFunctionality collectDeviceIdentifiers];
+    [RCCommonFunctionality setAdjustID:@""];
+    [RCCommonFunctionality setAppsflyerID:@""];
+    [RCCommonFunctionality setFBAnonymousID:@""];
+    [RCCommonFunctionality setMparticleID:@""];
+    [RCCommonFunctionality setOnesignalID:@""];
+    [RCCommonFunctionality setAirshipChannelID:@""];
+    [RCCommonFunctionality setMediaSource:@""];
+    [RCCommonFunctionality setCampaign:@""];
+    [RCCommonFunctionality setAdGroup:@""];
+    [RCCommonFunctionality setAd:@""];
+    [RCCommonFunctionality setKeyword:@""];
+    [RCCommonFunctionality setCreative:@""];
```

Comment:
All these should also be tested for nullable

Reply:
great call, adding